rust dhcp: Support DHCP options

cathay4t opened the pull request:
Support auto-dns, auto-gateway, auto-routes, auto-route-table-id.

Also support saving DNS configuration to interface with
`auto-dns: false`.

Integration test case enabled.

extra patches:
   * rust bond dummy: Fix bond and dummy DHCP info
   * rust route: support multipath route

cathay4t committed:
rust route: Add support of multipath route

For example, `ip -6 route show` could have

    default proto static metric 101 pref medium
            nexthop via 2001:db8:1::3 dev dhcpcli weight 1
            nexthop via fe80::20a7:aeff:fe2d:aa28 dev dhcpcli weight 1

Currently, nispor is ignoring it, this patch convert this to two route
entries.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
rust bond dummy: Fix bond and dummy DHCP info

Since NM plugin is treating bond and dummy as unknown interface which
cause their DHCP info not merged into nispor interface.

Fixed by assigning then correct interface type.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera commented on rust/src/lib/dns.rs:
```diff
@@ -261,7 +261,7 @@ pub(crate) fn reselect_dns_ifaces(
         find_ifaces_in_desire(true, &desired.interfaces)
             .or_else(|| {
-                find_ifaces_with_static_gateways(
+                find_ifaces_with_valid_for_dns(
```
nit: What about `find_valid_ifaces_for_dns`? I think the `with` is confusing on the function name.

ffmancera reviewed (approved):
LGTM! Thanks!

cathay4t committed:
rust dhcp: Support DHCP options

Support auto-dns, auto-gateway, auto-routes, auto-route-table-id.

Also support saving DNS configuration to interface with
`auto-dns: false`.

Integration test case enabled.

Signed-off-by: Gris Ge <fge@redhat.com>
